selfservice/hook/web_hook.go

```diff
@@ -307,9 +350,58 @@ func doHttpCall(method string, url string, as AuthStrategy, body io.Reader) erro
 
 	if err != nil {
 		return err
-	} else if resp.StatusCode >= 400 {
+	} else if resp.StatusCode >= http.StatusBadRequest {
```

Comment:
This actually has a connection leak. It should be written like so and also be using a resilient client:

```go
	client := httpx.NewResilientClient(/*...*/)
	resp, err := client.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()

	if resp.StatusCode >= http.StatusBadRequest {
```

You can instantiate the resilient client once in the hook struct `New` method.

Reply:
Done in 6b4c11081a322b71ff44231a9ead616580654442

Reply:
The connection leak is still there, reopening this. I have a local commit which fixes this.